src/mongo_types.ts

```diff
@@ -516,23 +520,23 @@ export type NestedPaths<Type> = Type extends
   ? [string]
   : Type extends object
   ? {
-      [Key in Extract<keyof Type, string>]: Type[Key] extends Type // type of value extends the parent
-        ? [Key]
-        : // for a recursive union type, the child will never extend the parent type.
-        // but the parent will still extend the child
-        Type extends Type[Key]
```

Comment:
This prevented template string keys from working because `` { a: { `example-${number}`: 1 } } extends { `example-${number}`: 1 } ``. This fact is exclusive to records with template string keys because template strings allow for infinite possible key values, which informs TypeScript that it should allow any keys on the object.